### PR TITLE
build: Bump Node.js to 16.19.1

### PR DESCRIPTION
**Pull request by Yakutoc**
Запланированный переход на версию  **Node.js 16.19.1**

Однако версия `NPM` **останется прежней** - **6.14.17**. 

Это связанно прежде всего с [изменениями в 7 версии NPM](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#changes-to-the-lockfile)

TL;DR

> Package-lock v2 and support for yarn.lock: Our new package-lock format will unlock the ability to do deterministically reproducible builds and includes everything npm will need to fully build the package tree. Prior to npm 7 yarn.lock files were ignored, the npm cli can now use yarn.lock as source of package metadata and resolution guidance.   

Переход на новую версию NPM будет отдельной задачей.  

 

:baby_chick: Download canary assets:

[colors--canary.427.4467285989.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.427.4467285989.xml)  
[color_metro_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_ios-swift--canary.427.4467285989.swift)  
[color_metro_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_kotlin--canary.427.4467285989.kt)  
[color_metro_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_metro_react-native--canary.427.4467285989.ts)  
[color_sberHealth_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.427.4467285989.swift)  
[color_sberHealth_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.427.4467285989.kt)  
[color_sberHealth_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.427.4467285989.ts)  
[color_sbermarket_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.427.4467285989.swift)  
[color_sbermarket_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.427.4467285989.kt)  
[color_sbermarket_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.427.4467285989.ts)  
[color_sberprime_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.427.4467285989.swift)  
[color_sberprime_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.427.4467285989.kt)  
[color_sberprime_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.427.4467285989.ts)  
[color_selgros_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_ios-swift--canary.427.4467285989.swift)  
[color_selgros_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_kotlin--canary.427.4467285989.kt)  
[color_selgros_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_selgros_react-native--canary.427.4467285989.ts)  
[color_smbusiness_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_ios-swift--canary.427.4467285989.swift)  
[color_smbusiness_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_kotlin--canary.427.4467285989.kt)  
[color_smbusiness_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_smbusiness_react-native--canary.427.4467285989.ts)  
[PlasmaTokensColor--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.427.4467285989.swift)  
[shadow_sbermarket_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.427.4467285989.ts)  
[typo_mage_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.427.4467285989.swift)  
[typo_mage_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.427.4467285989.kt)  
[typo_mage_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.427.4467285989.ts)  
[typo_plasma_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.427.4467285989.swift)  
[typo_plasma_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.427.4467285989.kt)  
[typo_plasma_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.427.4467285989.ts)  
[typo_ruler_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.427.4467285989.swift)  
[typo_ruler_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.427.4467285989.kt)  
[typo_ruler_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.427.4467285989.ts)  
[typo_sage_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.427.4467285989.swift)  
[typo_sage_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.427.4467285989.kt)  
[typo_sage_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.427.4467285989.ts)  
[typo_sbermarket_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.427.4467285989.swift)  
[typo_sbermarket_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.427.4467285989.kt)  
[typo_sbermarket_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.427.4467285989.ts)  
[typo_soulmate_ios-swift--canary.427.4467285989.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.427.4467285989.swift)  
[typo_soulmate_kotlin--canary.427.4467285989.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.427.4467285989.kt)  
[typo_soulmate_react-native--canary.427.4467285989.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.427.4467285989.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.163.1-canary.427.4467285989.0
  npm install @salutejs/plasma-temple@1.144.1-canary.427.4467285989.0
  # or 
  yarn add @salutejs/plasma-b2c@1.163.1-canary.427.4467285989.0
  yarn add @salutejs/plasma-temple@1.144.1-canary.427.4467285989.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-427/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-427/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-427/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-427/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-427/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-427/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-427/temple-storybook/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-427/

File: .npmrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 @salutejs:registry=https://registry.npmjs.org/
 
 save-exact=true
-  
+legacy-peer-deps=true
```
Comment (neretin-trike) [inline .npmrc:4]: а это сейчас нужно?

Comment (Yakutoc) [reply .npmrc:4]: Подложим соломку на всякий случай.

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-427/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-427/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-427/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-427/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-427/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-427/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-427/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-427/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-427/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-427/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-427/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-427/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-427/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-427/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-427/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-427/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-427/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-427/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-427/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-427/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: .npmrc (hunk 1 above)
Comment (Yeti-or) [reply .npmrc:6]: нужно иначе у нас будут ставиться и devDeps и peerDeps

File: .github/workflows/performance-test-pr.yml [hunk 2]
```diff
@@ -71,6 +72,7 @@ jobs:
 
       - name: Setup packages for PR
         run: |
+          npm install npm@6.14.17 -g
           npm ci
           npx lerna bootstrap
```
Comment (Yeti-or) [inline .github/workflows/performance-test-pr.yml:78]: что-то тут не так, там дальше шаг установки `16.15.x`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: Release
 
 on:
-  # Запуск в пулл-реквесте независимо от версионирования
+  # Запуск в pull request-e независимо от версионирования
```
Comment (Yeti-or) [inline .github/workflows/release.yml:4]: я бы не добавлял `-e` это же английское слово )

**Review by Yeti-or — APPROVED** [review]

File: .github/workflows/performance-test-pr.yml (hunk 2 above)
Comment (Yakutoc) [reply .github/workflows/performance-test-pr.yml:78]: @Yeti-or Ты про что именно? 

Если тут не ставить явно `npm@6.14.17`, будет ошибка, потому что другая версия npm. 

Снимок экрана 2023-03-20 в 16 44 49

Comment (Yeti-or) [reply .github/workflows/performance-test-pr.yml:78]: в этой джобе нужна другая версия node.js & npm похоже не важна какая, давай не будем по два раза устанавливать ноду ?

Comment (Yakutoc) [reply .github/workflows/performance-test-pr.yml:78]: @Yeti-or Не совсем тебя понял. 

[Тут валиться с оишбкой](https://github.com/salute-developers/plasma/actions/runs/4465097202/jobs/7841829732)

Comment (Yeti-or) [reply .github/workflows/performance-test-pr.yml:78]: я не понимаю зачем здесь 
Сначала [ставится node из nvm ( 16.19.1 )](https://github.com/salute-developers/plasma/actions/runs/4465097202/workflow#L24)
Делается `npm ci && lerna bootstrap`
Устанавливается [node 16.15.x ](https://github.com/salute-developers/plasma/actions/runs/4465097202/workflow#L44)
Запускается `npx perftool`
Опять [ставится node из nvm ( 16.19.1 )](https://github.com/salute-developers/plasma/actions/runs/4465097202/workflow#L61)
Снова `npm ci && lerna bootstrap`
Снова [node 16.15.x ](https://github.com/salute-developers/plasma/actions/runs/4465097202/workflow#L80)
Снова `npx perftool`


Зачем здесь 4 раза уставнавливать ноду и бутрапить зависимости ? 

cc @KateKate  может можно упростить? 
на сколько я понял идея чекаутить мастер а потом чекаут в ветку pr, 
но можно ли 1) уменьшить кол-во установок хотя бы ноды?
2) хранить артифакты например для мастера а не генерировать их на каждый пр?

Comment (Yeti-or) [reply .github/workflows/performance-test-pr.yml:78]: @Yakutoc смотря на твою ошибку я предполгаюа что версия npm все же важна, но внимательно логи не успеваю прочитать, ты понял теперь что я имею ввиду ?

Comment (Yakutoc) [reply .github/workflows/performance-test-pr.yml:78]: @Yeti-or Да, теперь понятно что у тебя вопрос к в целом как выстроен workflow у этого action.

**Comment by Yeti-or** [comment]
давай отдельной задачей уберем install node.js 16.x и заведем отдельную задачу на ускорение Component perfomance чтобы не было два lerna bootstrap – иначе очень долго

а это тогда вливаем as is

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-427/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: .npmrc (hunk 1 above)
Comment (neretin-trike) [reply .npmrc:6]: @Yeti-or поидее не должны, мы же на предыдущей версии npm остаёмся

**Review by neretin-trike — APPROVED** [review]

**Comment by Yeti-or** [comment]
🚢

**Pull request merged**